zbus/src/address/transport/unix.rs

```diff
@@ -1,129 +1,112 @@
-    /// A path to a unix domain socket on the filesystem.
-    File(PathBuf),
-    /// An abstract unix domain socket name.
-    #[cfg(target_os = "linux")]
-    Abstract(OsString),
-    /// A listenable address using the specified path, in which a socket file with a random file
-    /// name starting with 'dbus-' will be created by the server. See [UNIX domain socket address]
-    /// reference documentation.
-    ///
-    /// This address is mostly relevant to server (typically bus broker) implementations.
-    ///
-    /// [UNIX domain socket address]: https://dbus.freedesktop.org/doc/dbus-specification.html#transports-unix-domain-sockets-addresses
-    Dir(PathBuf),
-    /// The same as UnixDir, except that on platforms with abstract sockets, the server may attempt
-    /// to create an abstract socket whose name starts with this directory instead of a path-based
-    /// socket.
-    ///
-    /// This address is mostly relevant to server (typically bus broker) implementations.
-    TmpDir(PathBuf),
+impl<'a> Unix<'a> {
+    /// One of the various `unix:` addresses.
+    pub fn kind(&self) -> &UnixAddrKind<'a> {
+        &self.kind
+    }
 }
 
-impl Display for UnixSocket {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        fn fmt_unix_path(f: &mut Formatter<'_>, path: &OsStr) -> std::fmt::Result {
-            #[cfg(unix)]
-            {
-                use std::os::unix::ffi::OsStrExt;
+impl<'a> TryFrom<&'a Address<'a>> for Unix<'a> {
+    type Error = Error;
 
-                encode_percents(f, path.as_bytes())?;
+    fn try_from(s: &'a Address<'a>) -> Result<Self> {
+        let mut kind = None;
+        let mut iter = s.key_val_iter();
+        for (k, v) in &mut iter {
+            match k {
+                "path" | "dir" | "tmpdir" => {
+                    let v = v.ok_or_else(|| Error::MissingValue(k.into()))?;
+                    let v = decode_percents_os_str(v)?;
+                    kind = Some(match k {
+                        "path" => UnixAddrKind::Path(v),
+                        "dir" => UnixAddrKind::Dir(v),
+                        "tmpdir" => UnixAddrKind::Tmpdir(v),
+                        // can't happen, we matched those earlier
+                        _ => panic!(),
+                    });
+                    break;
+                }
+                "abstract" => {
+                    let v = v.ok_or_else(|| Error::MissingValue(k.into()))?;
+                    let v = decode_percents(v)?;
+                    kind = Some(UnixAddrKind::Abstract(v));
+                    break;
+                }
+                "runtime" => {
+                    let v = v.ok_or_else(|| Error::MissingValue(k.into()))?;
+                    let v = decode_percents_str(v)?;
+                    if v != "yes" {
+                        return Err(Error::InvalidValue(k.into()));
+                    }
+                    kind = Some(UnixAddrKind::Runtime);
+                    break;
```

Comment:
while you adress other things, can you please also add an empty line before the breaks here (and anywhere else you've `break` and `return` at the end of a block of code)? You can see the existing code style about that in the code you're deleting in this file at line 104 (I'm currently looking at the diff of the whole PR branch).